insights/client/__init__.py

```diff
@@ -253,10 +298,12 @@ def install(self, new_egg, new_egg_gpg_sig):
             raise
 
         # Copy the NEW (/tmp/insights-core.egg) egg to /var/lib/insights/newest.egg
+        # Additionally, copy NEW (/tmp/insights-core.egg.asc) to /var/lib/insights/newest.egg.asc
         try:
             logger.debug("Copying %s to %s." % (new_egg, constants.insights_core_newest))
             shutil.copyfile(new_egg, constants.insights_core_newest)
-            shutil.copyfile(new_egg_gpg_sig, constants.insights_core_gpg_sig_newest)
+            if config["gpg"] and new_egg_gpg_sig:
```

Comment:
I actually think it would be better to fetch and install the signature file even if gpg verification is currently disabled.  It might not be disabled next time around, thus invalidating the installed egg.